mantle/platform/machine/azure/flight.go

```diff
@@ -87,7 +87,15 @@ func (af *flight) NewCluster(rconf *platform.RuntimeConfig) (platform.Cluster, e
 	}
 
 	if !rconf.NoSSHKeyInMetadata {
-		ac.sshKey = af.SSHKey
+		// We have worked around the golang library limitation for
+		// keyexchange algorithm by switching to an ecdsa key in
+		// network/ssh.go.  However, Azure requires an RSA key.  For
+		// now (until we get an updated golang library) we'll just
+		// satisfy the requirement by using a fake key and disabling
+		// the fcos.ignition.misc.empty and fcos.ignition.v3.noop
+		// tests on Azure.
+		// https://docs.microsoft.com/en-us/azure/virtual-machines/linux/mac-create-ssh-keys#supported-ssh-key-formats
+		ac.sshKey = af.FakeSSHKey
 	} else {
 		ac.sshKey = af.FakeSSHKey
 	}
```

Comment:
I find this block kind of confusing (an if/else where both logic paths yield the same code). I think it would help if we actually comment out the original code:

```
// ac.sshKey = af.SSHKey
```

or otherwise mention (in the comment) that once the problem is fixed we should set this back to `ac.sshKey = af.SSHKey`. Also maybe we should link to https://github.com/coreos/coreos-assembler/issues/1772 here?